src/callbacks_step/save_restart_dg.jl

```diff
@@ -327,4 +327,28 @@ function load_restart_file_on_root(mesh::Union{ParallelTreeMesh, ParallelP4estMe
 
     return u_ode
 end
+
+function save_restart_controller(integrator,
+                                 controller, restart_callback)
+    # Save only on root
+    if mpi_isroot()
+        @unpack output_directory = restart_callback
+        timestep = integrator.stats.naccept
+
+        # Filename based on current time step
+        filename = joinpath(output_directory, @sprintf("restart_%06d.h5", timestep))
+
+        # Open file (preserve existing content)
+        h5open(filename, "cw") do file
+            # Add context information as attributes
+            attributes(file)["qold"] = integrator.qold
```

Comment:
Please prefix the values with something such that an unaware user has an idea of what this is all for. E.g., instead of
* `qold`
* `dtpropose`
* `controller_err`
use
* `time_integrator_qold`
* `time_integrator_dtpropose`
* `time_integrator_controller_err`